### PR TITLE
feat: Further refine how assertFor address prop works

### PR DESCRIPTION
**Pull request by chrismaree**
**Motivation**

When calling `assertTruth` previously if you wanted to be the asserter address you would set `address(0)` for the `asserter` field. This PR changes this such that you must set your own address to assert for yourself.
not that this PR was forked off of https://github.com/UMAprotocol/protocol/pull/4280 which should be merged before this is reviewed.

### CODE REVIEW

File: packages/core/contracts/optimistic-asserter/implementation/OptimisticAsserter.sol [hunk 1]
```diff
@@ -96,12 +95,11 @@ contract OptimisticAsserter is OptimisticAsserterInterface, Lockable, Ownable {
         uint256 liveness, // TODO: think about changing this to challenge window?
         bytes32 identifier
     ) public nonReentrant() returns (bytes32) {
-        // TODO: if you want to assert for yourself set this to your own address NOT address(0).
-        asserter = asserter == address(0) ? msg.sender : asserter;
         // TODO: think about placing either msg.sender or block.timestamp into the claim ID to block an advasery
         // creating a claim that collides with a known assertion that will be created into the future.
         bytes32 assertionId = _getId(claim, bond, liveness, currency, callbackRecipient, escalationManager, identifier);
 
+        require(asserter != address(0), "Asserter cant be 0");
```
Comment (Reinis-FRP) [inline packages/core/contracts/optimistic-asserter/implementation/OptimisticAsserter.sol:102]: Just to keep consistent interface we should also use the same logic in `disputeAssertionFor` by not allowing caller to provide 0 addrress.

Comment (chrismaree) [reply packages/core/contracts/optimistic-asserter/implementation/OptimisticAsserter.sol:102]: that is a great catch! I will update.

what do you think about re-naming `disputeAssertionFor` to just `disputeAssertion` now that we call `assertTruthFor` `assertTruth`?

Comment (Reinis-FRP) [reply packages/core/contracts/optimistic-asserter/implementation/OptimisticAsserter.sol:102]: yes, `disputeAssertion` sounds good!

File: packages/core/test/foundry/optimistic-asserter/OptimisticAsserter.EdgeCases.t.sol [hunk 2]
```diff
@@ -94,11 +94,11 @@ contract InvalidParameters is Common {
         defaultCurrency.approve(address(optimisticAsserter), defaultBond * 2);
 
         // Account2 disputes the assertion.
-        optimisticAsserter.disputeAssertionFor(assertionId, address(0));
+        optimisticAsserter.disputeAssertionFor(assertionId, TestAddress.account1);
```
Comment (Reinis-FRP) [inline packages/core/test/foundry/optimistic-asserter/OptimisticAsserter.EdgeCases.t.sol:97]: Better to use `TestAddress.account2` for disputes.

File: packages/core/test/foundry/optimistic-asserter/OptimisticAsserter.EdgeCases.t.sol [hunk 3]
```diff
@@ -94,11 +94,11 @@ contract InvalidParameters is Common {
...
 
         // Account2 should not be able to dispute the assertion again.
         vm.expectRevert("Assertion already disputed");
-        optimisticAsserter.disputeAssertionFor(assertionId, address(0));
+        optimisticAsserter.disputeAssertionFor(assertionId, TestAddress.account1);
```
Comment (Reinis-FRP) [inline packages/core/test/foundry/optimistic-asserter/OptimisticAsserter.EdgeCases.t.sol:101]: Better to use `TestAddress.account2` for disputes.

**Review by Reinis-FRP — APPROVED** [review]
looking good!

**Pull request merged**